src/factories/meta/CollectionCollector_factory.h

```diff
@@ -15,7 +15,7 @@ class CollectionCollector_factory : public JOmniFactory<CollectionCollector_fact
   std::unique_ptr<AlgoT> m_algo;
 
   typename JOmniFactory<CollectionCollector_factory<T>>::template VariadicPodioInput<T> m_inputs{
-      this};
+      this, std::vector<std::string>{""}, true};
```

Comment:
The optional nature of the collection is compile-time information. Maybe after the process of implementing this you have a good idea of what it would take to do this be entirely at compile-time.

Other question: are there no current (or potential) uses of CollectionCollector_factory where we do expect all collections to exist?

Reply:
> The optional nature of the collection is compile-time information. Maybe after the process of implementing this you have a good idea of what it would take to do this be entirely at compile-time.

I don't entirely follow what the distinction is here.

> Other question: are there no current (or potential) uses of CollectionCollector_factory where we do expect all collections to exist?

I can't think of anything, not to say it wouldn't arise in some way I haven't considered. I have an alternative implementation which has a SetOptional method in the InputBase which can be set in the CollectionCollector_factory::Configure as true, or via a non-existent CollectionCollectorConfig.

Reply:
I meant that we don't need to pass a bool variable and then query the bool variable with this->is_optional. This is all already known at compile time, so the bool could be a template parameter, and that can be used in the GetCollection call.